docs/kb/integrations/customize-semgrep-precommit.md

```diff
@@ -0,0 +1,71 @@
+---
+description: Understand how to customize Semgrep's behavior when using it with pre-commit.
+tags:
+  - Extensions
+---
+
+# Customize Semgrep in pre-commit
+
+Semgrep offers two standard pre-commit hooks [outlined here](/docs/extensions/overview/#pre-commit).
+
+* The `semgrep` hook is optimized to run with a provided set of rules, either local rules or rules from the Semgrep Registry.
+* The `semgrep-ci` hook is optimized to run with the rules from your organization in the Semgrep AppSec Platform, without sending results to the platform, since results from pre-commit are temporary and not linked to the final code in the repository.
```

Comment:
wonder if it's worth calling out that this is near identical to what happens when semgrep runs in CI on a PR specifically?

Reply:
Also, maybe we want to mention something about this requiring login?

Reply:
Plus on the login, minus on the "what happens when Semgrep runs on a PR" - it's actually significantly different from that since PR scans compare head and base, and pre-commit scans compare per-commit. We've had folks think that PR scans do per-commit analysis before and be confused that they don't, so I don't want to encourage the confusion.